### ISSUE TITLE
Number input component fails accessibility test

### ISSUE THREAD

**Issue by jefftmahoney** [IBM/carbon-components-angular#968]
## Detailed description

> Describe in detail the issue you're having.
When running the DAP plugin for Chrome, I'm seeing violations associated with the Number input control.  Specifically, it does not seem to like that the 'up' and 'down' arrows of the component, which are both buttons, do not have associated labels.

> Is this a feature request (new component, new icon), a bug, or a general issue?
Bug

> Is this issue related to a specific component?
Number input component

> What did you expect to happen? What happened instead? What would you like to see changed?
The component works fine - I would just like to see no violations when I run an accessibility-compliance tool like the DAP plugin for Chrome.

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
10.6.0 for `carbon-components` and 3.21.2 for `carbon-components-angular`

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
IBM Study Advance.  No pressing dates, but our OM indicates she would like to see us accessibility-compliant by late January.

## Steps to reproduce the issue

If you run this code (below) and fire up the DAP plugin for Chrome on it, you will see the problem.  ***Note***: the markup below was copied from the template for the Number input component.  I have simplified it somewhat to make it easier to review.  
***Possible Solution***: I have also added an `aria-label` to the first button (the 'up' button) to demonstrate a possible solution (if you run the DAP plugin, you'll see that this first button passes with no problems).  I realize that over-use of ARIA attributes can be a problem, but in this instance I feel it's justified, since adding actual `<label>` elements would be clunky, given the constraints of the template.

```
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>
</head>

<body>
    <main>
        <!-- <label *ngIf="skeleton && label" class="bx--label bx--skeleton"></label> -->
        <!-- <label [for]="id" class="bx--label">
            here is my label
        </label> -->
        <!-- <div *ngIf="helperText" class="bx--form__helper-text">
            <ng-container *ngIf="!isTemplate(helperText)">{{helperText}}</ng-container>
            <ng-template *ngIf="isTemplate(helperText)" [ngTemplateOutlet]="helperText"></ng-template>
        </div> -->
        <div data-numberinput [attr.data-invalid]="(invalid ? true : null)" class="bx--number">
            <div class="bx--number__input-wrapper">
                <label for="id" class="bx--label">
                    here is my label
                </label>
                <input type="number" id="id" [value]="value" [attr.min]="min" [attr.max]="max" [disabled]="disabled"
                    [required]="required" (input)="onNumberInputChange($event)" />
                <ibm-icon-warning-filled16 *ngIf="!skeleton && invalid" class="bx--number__invalid"
                    style="display: inherit;">
                </ibm-icon-warning-filled16>
                <div *ngIf="!skeleton" class="bx--number__controls">
                    <button class="bx--number__control-btn up-icon" type="button" aria-label="first button"
                        aria-live="polite" aria-atomic="true" (click)="onIncrement()">
                        <ibm-icon-caret-up16></ibm-icon-caret-up16>
                    </button>
                    <button class="bx--number__control-btn down-icon" type="button" aria-live="polite"
                        aria-atomic="true" (click)="onDecrement()">
                        <ibm-icon-caret-down16></ibm-icon-caret-down16>
                    </button>
                </div>
            </div>
            <div *ngIf="invalid" class="bx--form-requirement">
                <ng-container *ngIf="!isTemplate(invalidText)">{{invalidText}}</ng-container>
                <ng-template *ngIf="isTemplate(invalidText)" [ngTemplateOutlet]="invalidText"></ng-template>
            </div>
        </div>

    </main>
</body>

</html>
```

## Additional information

* Screenshots or code
* Notes

**Comment by cal-smith** [comment]
Thanks for catching this! adding `aria-label`s to the buttons is definitely the right approach in this case
